target/linux/realtek/files-5.15/drivers/net/ethernet/rtl838x_eth.c

```diff
@@ -2030,6 +2030,8 @@ static int rtl931x_mdio_reset(struct mii_bus *bus)
 	for (int i = 0; i < 56; i++) {
 		u32 pos;
 
+		if ((priv->smi_bus[i] < 0) || (priv->smi_bus[i] > MAX_SMI_BUSSES))
+			continue;
```

Comment:
Same problem with the first condition.

Reply:
Actually, this is a valid point, as we don't have to check for anything less then 0 :) so I can remove that first bit. as it was always a no-op; it won't hurt, but its just stupid indeed.

Reply:
`MAX_SMI_BUSSES` is also confusing to me. Is it the highest valid SMI bus index (`MAX_SMI_BUS`)? Or is it the number of valid SMI bus indices (`SMI_BUS_COUNT`)? If it's the latter, the second comparison is also wrong and should be "greater than or equal".

Reply:
Going from the message on the second patch, it appears 4 is the number of busses, as an index of 4 is invalid. So this also needs to be fixed.

Reply:
what do you mean with 'also'?

Reply:
The "greater than MAX_SMI_BUSSES" comparison.